subsys/power/device.c

```diff
@@ -133,7 +133,7 @@ void pm_resume_devices(void)
 	}
 }
 
-void pm_create_device_list(void)
+ __attribute__((optimize("O0"))) void pm_create_device_list(void)
```

Comment:
What is this for?

Reply:
That looks wrong. If you need no optimization you should set `CONFIG_NO_OPTIMIZATION` in the test.

Reply:
I tried. I found this problem when I run my cases with "--coverage" everything is OK, but cases fail without this option. When I realize that CONFIG_COVERAGE will enable CONFIG_NO_OPTIMIZATION, I enabled it in prj.conf, but got a "Booting from ROM..E: Double Fault" even before test cases get to run. So, here is my last choice.

Reply:
@pabigot Can you take look at these update?